<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <title>Test Global Error Handling</title>
  <!-- <script src="vue-grid-layout.umd.min.js"></script> -->
  <style>
    .empty {
      padding-top: 100%;
      background-color: red;
    }
  </style>
  <script>
    const timeout = (ms) =>
      new Promise((resolve, reject) => {
        setTimeout(() => {
          resolve();
        }, ms);
      });

    const ajax1 = () =>
      timeout(2000).then(() => {
        return 1;
      });

    const ajax2 = () =>
      timeout(1000).then(() => {
        return 2;
      });

    const ajax3 = () =>
      timeout(2000).then(() => {
        return 3;
      });

    // const mergePromise = (ajaxArray) => {
    //   // 在这里实现你的代码
    //   const res = [];

    //   for (let index = 0; index < ajaxArray.length; index++) {
    //     const ajax = ajaxArray[index];
    //     res[index] = ajax();
    //   }
    //   return Promise.all(res);
    // };

    const mergePromise = (ajaxArray) => {
      // Deal the promise like a chain
      let sequence = Promise.resolve();
      return new Promise((resolve) => {
        const result = [];

        for (let index = 0; index < ajaxArray.length; index++) {
          sequence = sequence.then(ajaxArray[index]).then((res) => {
            result.push(res)
            if (index === ajaxArray.length - 1) resolve(result);
          })
        }
      })
    };

    mergePromise([ajax1, ajax2, ajax3]).then((data) => {
      console.log('done');
      console.log(data); // data 为 [1, 2, 3]
    });
  </script>

</head>

<body>

  <h1>Test Global Error Handling with window.onerror</h1>
  <div class="empty"></div>
</body>

</html>